cryptarithm.py

```diff
@@ -13,135 +13,111 @@
 # limitations under the License.
 
 import argparse
-from typing import List
-from pprint import pprint
 from collections import defaultdict
-from itertools import combinations
-import numpy as np
+from typing import List
 
-from dimod import DQM
-from dwave.system import LeapHybridDQMSampler
-from utilities import parse_problem_file, update_coefficient_map_and_first_letter_set, render_solution
+from dimod import Binary, CQM, Integer, quicksum
+from dwave.system import LeapHybridCQMSampler
+from utilities import (parse_problem_file, 
+                       update_coefficient_map_and_first_letter_set, 
+                       render_solution)
 
-class LetterVariable:
-    """Class for holding information about variables in alphametics problems.
+class ModelVariable:
+    """Container for model variable information.
 
     Args:
-        name: Variable string id.
-        coefficient: Coefficient on variable from formulation.
-        first_letter: Whether or not variable is first letter of a word.
-
+        label: Single letter label for the model variable.
+        coefficient: Term that appears in main equality constraint.
+        first_letter: Whether or not letter is a first letter.
+    
     """
-    def __init__(self, name:str = None, coefficient:int = 0, first_letter:bool = False):
-        self.name = name
+    def __init__(self, label: str, coefficient: float, first_letter: bool = False):
+        self.label = label
         self.coefficient = coefficient
         self.first_letter = first_letter
+    
         if self.first_letter:
-            self.domain = tuple(range(1,10))
+            self.var = Integer(self.label, lower_bound=1, upper_bound=9)
+            self.lower_bound = 1
+            self.upper_bound = 9
         else:
-            self.domain = tuple(range(10))
+            self.var = Integer(self.label, lower_bound=0, upper_bound=9)
+            self.lower_bound = 0
+            self.upper_bound = 9
 
-    def __repr__(self):
-        return "(name: {name}, coefficient: {coeff}, domain: {domain})".format(name=self.name,
-                                                                               coeff=self.coefficient,
-                                                                               domain=self.domain)
 
-
-def build_dqm(variable_list: List[LetterVariable], coefficient_map: dict) -> DQM:
-    """Build the discrete quadratic model from provided variable list.
+def build_cqm(model_variables: List[ModelVariable]) -> CQM:
```

Comment:
It's a minor thing, but IMO is customer-facing code we should spell out `Constrained Quadratic Model`
```suggestion
def build_cqm(model_variables: List[ModelVariable]) -> ConstrainedQuadraticModel:
```